pkg/apis/kubermatic/v1/datacenter.go

```diff
@@ -642,6 +642,9 @@ type DatacenterSpecOpenstack struct {
 	// Optional: configures enablement of topology support for the Cinder CSI Plugin.
 	// This requires Nova and Cinder to have matching availability zones configured.
 	CSICinderTopologyEnabled bool `json:"csiCinderTopologyEnabled,omitempty"`
+	// Optional: enable a configuration drive that will be attached to the instance when it boots.
+	// The instance can mount this drive and read files from it to get information
+	EnableConfigDrive *bool `json:"enableConfigDrive,omitempty"`
```

Comment:
I would instead name this `enforceConfigDrive` and use a plain boolean instead of a boolean pointer. The option to use `config drive` for machine deployments will be exposed via the dashboard and one of the requirements from the customer was to "enforce" config drive and disable overrides from UI.